api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -242,9 +249,9 @@ public static Cohort intersect(Cohort a, Cohort b) {
 	public static Cohort subtract(Cohort a, Cohort b) {
 		Cohort ret = new Cohort();
 		if (a != null) {
-			ret.getMemberIds().addAll(a.getMemberIds());
+			ret.getMembers().addAll(a.getMembers());
 			if (b != null) {
-				ret.getMemberIds().removeAll(b.getMemberIds());
+                ret.getMembers().removeAll(b.getMembers());
```

Comment:
I doubt if you implemented the equals and hashcode methods in CohortMember, calls like this that depend on the equals method will most likely misbehave since equals compares patient objects (i.e  their uuids) but instances of patient created with new Patient(id) wont have a matching uuid. You might want to implement the hashcode and equals methods in CohortMember class